src/pykiso/lib/connectors/cc_process.py

```diff
@@ -0,0 +1,319 @@
+##########################################################################
+# Copyright (c) 2010-2022 Robert Bosch GmbH
+# This program and the accompanying materials are made available under the
+# terms of the Eclipse Public License 2.0 which is available at
+# http://www.eclipse.org/legal/epl-2.0.
+#
+# SPDX-License-Identifier: EPL-2.0
+##########################################################################
+
+"""
+Process Channel
+***************
+
+:module: cc_process
+
+:synopsis: CChannel implementation for process execution.
+
+The CCProcess channel provides functionality to start a process and
+to communicate with it.
+
+.. currentmodule:: cc_process
+
+"""
+
+import logging
+import queue
+import subprocess
+import threading
+from dataclasses import dataclass
+from typing import IO, ByteString, Callable, Dict, List, Optional, Tuple, Union
+
+from pykiso import Message
+from pykiso.connector import CChannel
+
+log = logging.getLogger(__name__)
+
+MessageType = Union[str, ByteString]
+
+
+class CCProcessError(BaseException):
+    ...
+
+
+@dataclass
+class ProcessMessage:
+    """Holds the data that is read from the process"""
+
+    # Stream name: stdout or stderr
+    stream: str
+    # Data as bytes or string
+    data: Union[str, bytes]
+
+
+class ProcessExit:
+    ...
+
+
+# Marker for process exit in queue_in
+PROCESS_EXIT = ProcessExit()
```

Comment:
Maybe make this private 
```suggestion
_PROCESS_EXIT_SENTINEL = ProcessExit()
```
Also, why making an additional ProcessExit class when you can just do this:
```python
_PROCESS_EXIT_SENTINEL = object()
```

Reply:
Because then the type of the queue items is Any, which I did not like. Or does Union[ProcessMessage, object] make sense? Because everything is object?